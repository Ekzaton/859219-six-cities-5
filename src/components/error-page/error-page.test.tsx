import React from 'react';
import {MemoryRouter} from 'react-router-dom';
import {render} from '@testing-library/react';

import ErrorPage from "./error-page";

describe(`Error page renders correctly`, () => {
  it(`for loading error`, () => {
    const loadingError = {
      data: {error: `Sorry service is not available!`},
      status: 503,
      statusText: `Temporarily Unavailable`,
      headers: [],
      config: {}
    };

    const {container} = render(
        <MemoryRouter>
          <ErrorPage
            loadingError={loadingError}
          />
        </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });

  it(`for unknown route`, () => {
    const {container} = render(
        <MemoryRouter>
          <ErrorPage/>
        </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });
});
