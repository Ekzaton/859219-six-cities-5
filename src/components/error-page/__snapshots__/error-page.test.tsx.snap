// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error page renders correctly for loading error 1`] = `
<div>
  <main
    class="page__main"
  >
    <h1
      class="property__name"
      style="margin-top: 35vh;"
    >
      Sorry service is not available!
    </h1>
    <p
      class="property__inside-title"
      style="margin: 5vh 0px 20vh;"
    >
      Error 503: Temporarily Unavailable
    </p>
    <a
      href="/"
    >
      <p
        class="property__text"
        style="text-align: center; text-decoration: underline;"
      >
        Back to main page
      </p>
    </a>
  </main>
</div>
`;

exports[`Error page renders correctly for unknown route 1`] = `
<div>
  <main
    class="page__main"
  >
    <h1
      class="property__name"
      style="margin-top: 35vh;"
    >
      Sorry cant find that!
    </h1>
    <p
      class="property__inside-title"
      style="margin: 5vh 0px 20vh;"
    >
      Error 404: Not Found
    </p>
    <a
      href="/"
    >
      <p
        class="property__text"
        style="text-align: center; text-decoration: underline;"
      >
        Back to main page
      </p>
    </a>
  </main>
</div>
`;
